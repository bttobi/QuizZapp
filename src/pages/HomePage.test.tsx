/**
 * @jest-environment jsdom
 */
import { HomePage } from '.';
import { render, waitFor } from '../tests/test-utils';

const setup = () => render(<HomePage />);

describe('HomePage component', () => {
  it('displays the HomePage content', async () => {
    const { getByText, getByRole } = setup();
    await waitFor(() => {
      expect(getByRole('button')).toBeDefined();
      expect(
        getByText(/Welcome to our dynamic quiz creation platform - QuizZapp!/i)
      ).toBeDefined();
      expect(getByRole('img')).toBeDefined();
    });
  });

  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
