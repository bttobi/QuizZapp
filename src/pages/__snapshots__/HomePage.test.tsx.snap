// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage component > matches snapshot 1`] = `
<div>
  <div
    class="w-full h-full undefined"
    style="opacity: 0;"
  >
    <section
      class="flex gap-20 items-center h-full"
    >
      <article
        class="flex sm:flex-row sm:text-x3l flex-col justify-center align-center items-center gap-10 w-auto bg-primary/60 p-4 rounded-lg mx-10 text-md text-justify shadow-lg"
      >
        <div
          class="sm:scale-125 scale-100 ml-4 w-full h-full"
          style="transform: translateY(5rem) translateZ(0);"
        >
          <img
            class="mr-4 + sm:scale-125 scale-100 ml-4"
            height="128px"
            src="https://private-user-images.githubusercontent.com/76923032/279826261-22cfd9bc-5391-4feb-be33-a294b3b0603f.svg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUyNTE0OTMsIm5iZiI6MTcwNTI1MTE5MywicGF0aCI6Ii83NjkyMzAzMi8yNzk4MjYyNjEtMjJjZmQ5YmMtNTM5MS00ZmViLWJlMzMtYTI5NGIzYjA2MDNmLnN2Zz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE0VDE2NTMxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU4NzFkYTZhMzk4NmM1NDlkYmU5MDIzNTM4NzQ1NjFjMmNhMjY1OWRmZjI4NTA1M2FkZTlhZjY4N2E2ZWQ4M2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.iJj94shtiBOC9OCHaZHEWHGfGBZbkuzhMhIkbzK07tc"
            width="128px"
          />
        </div>
        <p
          class="flex flex-col justify-center items-center mx-2 sm:text-2xl text-xl"
        >
          <span
            style="transform: translateX(5rem) translateZ(0);"
          >
            Welcome to our dynamic quiz creation platform - QuizZapp! Unleash your inner quizmaster and craft captivating quizzes on any topic under the sun. Dive into a world of knowledge, challenge your friends, and climb the ranks on our leaderboard by answering questions with precision. Earn points for every correct answer, unlocking new levels of expertise as you conquer diverse trivia. Join us in this interactive journey where learning meets excitement and test your wits against the best!
          </span>
          <button
            class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-unit-4 min-w-unit-20 h-unit-10 gap-unit-2 rounded-medium [&>svg]:max-w-[theme(spacing.unit-8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none bg-success data-[hover=true]:opacity-hover mt-8 w-48 animate-pulse text-white text-xl"
            type="button"
          >
            Get started now!
          </button>
        </p>
      </article>
    </section>
  </div>
</div>
`;
